test/rubygems/test_gem_remote_fetcher.rb

```diff
@@ -606,9 +606,13 @@ def fetcher.request(uri, request_class, last_modified = nil)
   def test_fetch_s3
     fetcher = Gem::RemoteFetcher.new nil
     @fetcher = fetcher
-    url = 's3://testuser:testpass@my-bucket/gems/specs.4.8.gz'
```

Comment:
We probably shouldn't break compatibility for people doing it this way.. but I agree your method is probably better.

Still we would need to support both ways of authenticating to S3 for the time being.

Could you update your patch to support both?